server/src/main/java/org/opensearch/cluster/metadata/MetadataCreateIndexService.java

```diff
@@ -557,7 +557,7 @@ IndexMetadata buildAndValidateTemporaryIndexMetadata(
 
         // remove the setting it's temporary and is only relevant once we create the index
         final Settings.Builder settingsBuilder = Settings.builder().put(aggregatedIndexSettings);
-        settingsBuilder.remove(IndexMetadata.INDEX_NUMBER_OF_ROUTING_SHARDS_SETTING.getKey());
```

Comment:
Guess this is intended, `index.number_of_routing_shards` couldn't exist in the settings of IndexMetadata because it's a static setting, can only be set when creating index, so we added a new field `routing_num_shards` outside the index settings to the IndexMetadata, change this may cause unexpected behavior.

Reply:
If removing this seems risky, I have another [draft solution](https://github.com/opensearch-project/OpenSearch/pull/14446) that doesn't require removing this line. Would it be better to go forward with this approach instead?

Reply:
I think the solution in the draft PR is better.

Reply:
Oh okay sounds good! I will get that PR ready and close this one out!